File: 5.最长回文子串.cc
#include <string>
using std::string;

class Solution {
public:
    string longestPalindrome(string s) {
        if (s.empty())
            return "";
        string ans = s.substr(0, 1);
        for (int i = 0; i < s.length(); ++i) {
            string tmp = midToSize(s, i, i);
            if (tmp.length() > ans.length()) {
                ans = tmp;
            }
            tmp = midToSize(s, i, i + 1);
            if (tmp.length() > ans.length()) {
                ans = tmp;
            }
        }
        return ans;
    }

    string midToSize(string s, int left, int right) {
        while (left >= 0 && right < s.length() && s[left] == s[right]) {
            --left;
            ++right;
        }
        return s.substr(left + 1, right - left - 1);
    }
};

File: Golang/5.最长回文子串.go
package leetcode

func longestPalindrome(s string) string {
	if len(s) == 0 {
		return ""
	}
	ans := s[0:1]
	for i := 0; i < len(s)-1; i++ {
		tmp := midToSide(s, i, i)
		if len(tmp) > len(ans) {
			ans = tmp
		}
		tmp = midToSide(s, i, i+1)
		if len(tmp) > len(ans) {
			ans = tmp
		}
	}
	return ans

}

func midToSide(s string, left int, right int) string {
	for left >= 0 && right < len(s) && s[left] == s[right] {
		left--
		right++
	}
	return s[left+1 : right]
}
